### PR TITLE
Issue 660: Refactor the UserController to use form validation

### PR DESCRIPTION
**Pull request by ptirador**
Related issue: #660

Methods modified in `UserController` to follow the controllers guide and to use form validation.

The proper tests cases have been modified and new cases have been added.

### CODE REVIEW

**Comment by steve-todorov** [comment]
@ptirador this looks ready for a review. Is there anything more you need to do or should I test it out? :)

**Comment by ptirador** [comment]
It's allí yours!

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/forms/users/UserForm.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        return accessModel;
+    }
+
+    public void setAccessModel(AccessModelForm accessModel)
+    {
+        this.accessModel = accessModel;
+    }
+
+    public interface NewUser
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/forms/users/UserForm.java]: What is the the purpose of these two interfaces?

Comment (carlspring) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/forms/users/UserForm.java]: Thanks for the explanation!

**Comment by carlspring** [comment]
Thanks for these changes!

**Pull request merged**